java/src/main/java/io/cucumber/gherkin/GherkinLine.java

```diff
@@ -139,11 +163,11 @@ List<GherkinLineSpan> parseTableCells() {
                         // Skip the first empty span
                         beforeFirst = false;
                     } else {
-                        Entry<String, Integer> trimmedCellIndent = trimAndIndentKeepNewLines(cellBuilder.toString());
-                        int column = indent + cellStart + trimmedCellIndent.getValue() + COLUMN_OFFSET;
-                        lineSpans.add(new GherkinLineSpan(column, trimmedCellIndent.getKey()));
+                        StringUtils.IndentedText trimmedCellIndent = trimAndIndentKeepNewLines(cellBuilder.toString());
```

Comment:
Same as `parseTags`, `parseTableCells` depends on a condition in the matcher. Extract out into `TableLineParser.parse(int indent, String text)`.